backend/spec/features/admin/homepage_spec.rb

```diff
@@ -73,8 +73,6 @@
     custom_authorization! do |_user|
       can [:admin, :home], :dashboards
       can [:admin, :edit, :index, :show], Spree::Order
-      cannot [:show], Spree::StockLocation
-      cannot [:show], Spree::Zone
```

Comment:
Shouldn't these two be changed to:

```rb
      cannot [:admin], Spree::StockLocation
      cannot [:admin], Spree::Zone
```

So we can be sure that the settings tab is still visible even when the user is not able to access some of its sub-resources?

Reply:
Even with the suggested change, the settings tab is not visible. I could change it this way to make sure the settings tab is displayed if one of the inner tab can be accessed but this won't test that the rest of the settings are excluded:

```ruby
context 'as fakedispatch user' do
    before do
      allow_any_instance_of(Spree::Admin::BaseController).to receive(:try_spree_current_user).and_return(nil)
    end

    custom_authorization! do |_user|
      can [:admin, :home], :dashboards
      can [:admin, :edit, :index, :show], Spree::Order
      can [:admin], Spree::Zone
    end

    it 'should only display tabs fakedispatch has access to' do
      visit spree.admin_path
      expect(page).to have_link('Orders')
      expect(page).not_to have_link('Products')
      expect(page).not_to have_link('Promotions')
      expect(page).to have_link('Settings')
    end
  end
```

Let me think about something to test that.

Reply:
@elia maybe I've got a possible test, let me know your thoughts!

Reply:
Thanks @kennyadsl this look like a nice canary to catch if anything is going wrong with permissions without checking all the possible combinations 👍